consensus/polybft/block_builder.go

```diff
@@ -0,0 +1,290 @@
+)
+
+// TODO: Add opentracing
+
+// type txStatus uint8
+
+// const (
+// 	success txStatus = iota
+// 	fail
+// 	skip
+// 	nothing
+// )
+
+// type txPoolInterface interface {
+// 	Prepare()
+// 	Length() uint64
+// 	Peek() *types.Transaction
+// 	Pop(tx *types.Transaction)
+// 	Drop(tx *types.Transaction)
+// 	Demote(tx *types.Transaction)
+// 	ResetWithHeaders(headers ...*types.Header)
+// 	SetSealing(bool)
+// }
+
+// type txEvmTransition interface {
+// 	Write(txn *types.Transaction) error
+// 	WriteFailedReceipt(txn *types.Transaction) error
+// }
+
+// BlockBuilderParams are fields for the block that cannot be changed
+type BlockBuilderParams struct {
+	// Parent block
+	Parent *types.Header
+
+	Executor *state.Executor
+
+	// Coinbase that is signing the block
+	Coinbase types.Address
+
+	// ChainConfig is the configurtion of the chain
+	ChainConfig *chain.Params
+
+	// ChainContext interface is used during EVM execution
+	//ChainContext core.ChainContext
+
+	// Vanity extra for the block
+	Extra []byte
+
+	// GasLimit is the gas limit for the block
+	GasLimit uint64
+
+	// duration for one block
+	BlockTime time.Duration
+
+	// Logger
+	//Logger hcf.Logger
+
+	// TxPool txPoolInterface // Reference to the transaction pool
+}
+
+func NewBlockBuilder(params *BlockBuilderParams) *BlockBuilder {
+	// extra can only be 32 size max. it is better to trim that to return
+	// an error that we have to propagate. It should be up to higher level
+	// code to error if the extra supplied by the user is too big.
+	if len(params.Extra) > signer.IstanbulExtraVanity {
+		params.Extra = params.Extra[:signer.IstanbulExtraVanity]
+	}
+
+	if params.Extra == nil {
+		params.Extra = make([]byte, 0)
+	}
+
+	if params.BlockTime == 0 {
+		params.BlockTime = time.Second * 2 // TODO: is this ok?
+	}
+
+	builder := &BlockBuilder{
+		params: params,
+	}
+	builder.Reset()
+
+	return builder
+}
+
+var _ blockBuilder = &BlockBuilder{}
+
+type BlockBuilder struct {
+	// input params for the block
+	params *BlockBuilderParams
+
+	// header is the header for the block being created
+	header *types.Header
+
+	// transactions are the data included in the block
+	txns []*types.Transaction
+
+	// block is a reference to the already built block
+	block *types.Block
+
+	state *state.Transition
+}
+
+// Reset is used to indicate that the current block building has been interrupted
+// and it has to clean any data
+func (b *BlockBuilder) Reset() {
+	b.header = &types.Header{
+		ParentHash:   b.params.Parent.Hash,
+		Number:       b.params.Parent.Number + 1,
+		Miner:        b.params.Coinbase[:],
+		Difficulty:   1,
+		ExtraData:    b.params.Extra,
+		StateRoot:    types.EmptyRootHash, // this avoids needing state for now
+		TxRoot:       types.EmptyRootHash,
+		ReceiptsRoot: types.EmptyRootHash, // this avoids needing state for now
+		Sha3Uncles:   types.EmptyUncleHash,
+		GasLimit:     b.params.GasLimit, //will need to adjust dynamically later.
```

Comment:
Nitpick: please have a space between `//` and the rest of the comment. Additionally, comments should not end in periods